components/devicecloud-api/org.wso2.carbon.device.mgt.iot.common.api/src/main/java/org/wso2/carbon/device/mgt/iot/common/api/GroupManagerService.java

```diff
@@ -0,0 +1,252 @@
+package org.wso2.carbon.device.mgt.iot.common.api;
+
+import org.wso2.carbon.device.mgt.common.Device;
+import org.wso2.carbon.device.mgt.common.DeviceIdentifier;
+import org.wso2.carbon.device.mgt.group.common.Group;
+import org.wso2.carbon.device.mgt.group.common.GroupManagementException;
+import org.wso2.carbon.device.mgt.iot.common.GroupManagement;
+import org.wso2.carbon.user.core.Permission;
+
+import javax.ws.rs.*;
+import java.util.Date;
+import java.util.List;
+
+public class GroupManagerService {
+
+    @Path("/group")
+    @POST
+    @Consumes("application/json")
+    @Produces("application/json")
+    public boolean addGroup(@FormParam("name") String name, @FormParam("username") String username, @FormParam("description") String description) {
+        Group group = new Group();
+        group.setName(name);
+        group.setDescription(description);
+        group.setOwnerId(username);
+        group.setDateOfCreation(new Date().getTime());
+        group.setDateOfLastUpdate(new Date().getTime());
+        try {
+            return new GroupManagement().getGroupManagementService().createGroup(group);
+        } catch (GroupManagementException e) {
+            e.printStackTrace();
+            return false;
+        }
+    }
+
+    @Path("/group/id/{groupId}")
+    @PUT
+    @Consumes("application/json")
+    @Produces("application/json")
+    public boolean updateGroup(@PathParam("groupId") int groupId, @FormParam("name") String name, @FormParam("username") String username, @FormParam("description") String description) {
+        Group group = new Group();
+        group.setId(groupId);
+        group.setName(name);
+        group.setDescription(description);
+        group.setOwnerId(username);
+        group.setDateOfLastUpdate(new Date().getTime());
+        try {
+            new GroupManagement().getGroupManagementService().updateGroup(group);
+            return true;
+        } catch (GroupManagementException e) {
+            e.printStackTrace();
+            return false;
+        }
+    }
+
+    @Path("/group/id/{groupId}")
+    @DELETE
+    @Consumes("application/json")
+    @Produces("application/json")
+    public boolean deleteGroup(@PathParam("groupId") int groupId, @FormParam("username") String username) {
```

Comment:
Ack, Already done in latest repository. Code moved to https://github.com/wso2/carbon-device-mgt